src/jit/RegisterAlloc.cpp

```diff
@@ -87,26 +106,43 @@ class RegisterSet {
 
 class RegisterFile {
 public:
-    RegisterFile(uint32_t numberOfIntegerScratchRegs, uint32_t numberOfIntegerSavedRegs,
-                 uint32_t numberOfFloatScratchRegs, uint32_t numberOfFloatSavedRegs)
+    RegisterFile(uint32_t numberOfIntegerScratchRegs, uint32_t numberOfIntegerSavedRegs)
         : m_integerSet(numberOfIntegerScratchRegs, numberOfIntegerSavedRegs, true)
-        , m_floatSet(numberOfFloatScratchRegs, numberOfFloatSavedRegs, false)
+        , m_floatSet(SLJIT_NUMBER_OF_SCRATCH_FLOAT_REGISTERS, SLJIT_NUMBER_OF_SAVED_FLOAT_REGISTERS, false)
+#if (defined SLJIT_SEPARATE_VECTOR_REGISTERS && SLJIT_SEPARATE_VECTOR_REGISTERS)
+#if (defined SLJIT_CONFIG_RISCV && SLJIT_CONFIG_RISCV)
+        , m_vectorSet(SLJIT_NUMBER_OF_SCRATCH_VECTOR_REGISTERS - 1, SLJIT_NUMBER_OF_SAVED_VECTOR_REGISTERS, false)
+#else /* !SLJIT_CONFIG_RISCV */
+        , m_vectorSet(SLJIT_NUMBER_OF_SCRATCH_VECTOR_REGISTERS, SLJIT_NUMBER_OF_SAVED_VECTOR_REGISTERS, false)
+#endif /* SLJIT_CONFIG_RISCV */
+#endif /* SLJIT_SEPARATE_VECTOR_REGISTERS */
     {
     }
 
     RegisterSet& integerSet() { return m_integerSet; }
     RegisterSet& floatSet() { return m_floatSet; }
-
+    // clang-format off
```

Comment:
Is the format is that bad? Maybe we could turn all one liners to three lines, where the { and } in different line.